### PR TITLE
[Driver][SYCL] Enable multi-file for AOT

### PR DESCRIPTION
**Pull request by mdtoguchi**
When dealing with AOT compilations, we explicitly turned off multi-file
processing.  This behavior was preventing kernel name table generation.
This effectively makes multi-file the default for all modes.

### CODE REVIEW

**Review by AGindinson — COMMENTED** [review]
The implementation LGTM - still, I'd appreciate if @kbobrovs could take a look as well.

File: clang/lib/Driver/Driver.cpp [hunk 1]
```diff
@@ -3785,9 +3785,6 @@ class OffloadingActionBuilder final {
     /// Flag to signal if the user requested device code split.
     bool DeviceCodeSplit = false;
 
-    /// Flag to signal if DAE optimization is turned on.
```
Comment (kbobrovs) [inline clang/lib/Driver/Driver.cpp:3788]: Is DAE handling elimination related to this change? Not a blocker - just asking :)

File: clang/lib/Driver/Driver.cpp [hunk 2]
```diff
@@ -4342,10 +4339,8 @@ class OffloadingActionBuilder final {
         ActionList WrapperInputs;
         // post link is not optional - even if not splitting, always need to
         // process specialization constants
-        bool MultiFileActionDeps = !isSpirvAOT || DeviceCodeSplit || EnableDAE;
-        types::ID PostLinkOutType = isNVPTX || !MultiFileActionDeps
-                                        ? types::TY_LLVM_BC
-                                        : types::TY_Tempfiletable;
+        types::ID PostLinkOutType =
```
Comment (kbobrovs) [inline clang/lib/Driver/Driver.cpp:4342]: please update the [picture and the doc](https://github.com/intel/llvm/blob/sycl/sycl/doc/CompilerAndRuntimeDesign.md#device-code-post-link-step)

File: clang/lib/Driver/Driver.cpp (hunk 1 above)
Comment (mdtoguchi) [reply clang/lib/Driver/Driver.cpp:3788]: It's not related - but given the change of the default - the `EnableDAE` check became irrelevant, so I cleaned it up.

Comment (kbobrovs) [reply clang/lib/Driver/Driver.cpp:3788]: I see now. Thanks!

**Review by AGindinson — DISMISSED** [review]

**Pull request merged**